app/controllers/revisions_controller.rb

```diff
@@ -46,13 +46,21 @@ def trash
   end
 
   def recover
-    version = RecoverableVersion.find(params[:id])
-    class_name = object.class.name.humanize
+    version    = RecoverableVersion.find(params[:id])
+    class_name = if version.object.is_a?(Note) && version.object.node_id == Node.issue_library.id
```

Comment:
I hear you, on the one hand, it is painful to know it's there. On the other, it's difficult to justify prioritising this over other stuff when it isn't really working, and some of the other fish we need to fry are a lot bigger.